### PR TITLE
fix(engine): updates for PyJWT 2.1.0 compatibility 

### PR DESCRIPTION
**Pull request by pcanto-hopeit**

### CODE REVIEW

File: engine/requirements.txt [hunk 1]
```diff
@@ -6,8 +6,8 @@ dataclasses-jsonschema[fast-validation]
 stringcase
 aioredis
 lz4
-cryptography>=3.3.1,<3.4
-PyJWT[crypto]>=1.7.1,<2
+cryptography<4,>=3.4.7
+PyJWT[crypto]<3,>=2.1.0
```
Comment (leosmerling-hopeit) [inline engine/requirements.txt:10]: I think the format should be >=2.1.0,<3 to ensure setup.py parses correctyly

File: engine/src/hopeit/toolkit/auth.py [hunk 2]
```diff
@@ -141,14 +141,14 @@ def app_private_key(app_key: str) -> RSAPrivateKey:
 
 def new_token(app_key: str, payload: dict) -> str:
     private_key = app_private_key(app_key)
-    token = jwt.encode(
+    token = str(jwt.encode(
```
Comment (leosmerling-hopeit) [inline engine/src/hopeit/toolkit/auth.py:144]: if jwt.encode returns bytes the right way to convert to string is using decode()

File: engine/src/hopeit/toolkit/auth.py [hunk 3]
```diff
@@ -141,14 +141,14 @@ def app_private_key(app_key: str) -> RSAPrivateKey:
...
         {**payload, "app": app_key}, private_key, algorithm='RS256'
-    )
-    return token.decode()
+    ))
+    return token
 
 
 def decode_token(token: str) -> dict:
-    info = jwt.decode(token, verify=False)
+    info = jwt.decode(token, algorithms=['RS256'], options={"verify_signature": False})
```
Comment (leosmerling-hopeit) [inline engine/src/hopeit/toolkit/auth.py:151]: Are you sure a dict is needed for options or can they be sent as **kwargs?

File: plugins/auth/basic-auth/setup.py [hunk 4]
```diff
@@ -22,7 +22,7 @@
     python_requires=">=3.7",
     install_requires=[
         f"hopeit.engine=={version['ENGINE_VERSION']}",
-        "PyJWT[crypto]>=1.7.1,<2"
+        "PyJWT[crypto]<3,>=2.1.0"
```
Comment (leosmerling-hopeit) [inline plugins/auth/basic-auth/setup.py:25]: Reverse >=, <

**Review by leosmerling-hopeit — CHANGES_REQUESTED** [review]

File: engine/src/hopeit/toolkit/auth.py (hunk 2 above)
Comment (pcanto-hopeit) [reply engine/src/hopeit/toolkit/auth.py:144]: It returns str in the new version, the str casting is not needed.-

**Review by pcanto-hopeit — COMMENTED** [review]
self review

**Review by leosmerling-hopeit — APPROVED** [review]

**Pull request merged**